pkgs/shells/rc/default.nix

```diff
@@ -0,0 +1,48 @@
+{ stdenv, fetchurl, autoreconfHook
+, ncurses #acinclude.m4 wants headers for tgetent().
+, historySupport ? false
+, editSupport ? "readline", readline ? null, libedit ? null
+}:
+
+assert editSupport == "readline" -> readline != null;
```

Comment:
I suppose it's a question of style, but this could be simplified by only specifying `readline` and `libedit` and make the choice of edit support be implicit. That is, setting `readline` to non-null makes it the preferred choice, whereas setting neither would imply no edit support. With luck, you could then rely on the configure script to figure out which library to support based on what it finds in the environment, simplifying the expression further.

Some people may find your approach more readable (and it is a common way of doing things), but to me at least it's a lot of unnecessary verbiage.

More importantly, I wonder what the use case is for making this configurable at all; if there is a preferred alternative, use it. I could see using libedit on *BSD and readline on GNU/Linux, but that would be an internal detail.